queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/shares.py

```diff
@@ -71,6 +71,73 @@ def update_share_structure(business: Business, share_structure: Dict) -> Optiona
     return err
 
 
+def update_share_structure_correction(business: Business, share_structure: Dict) -> Optional[List]:
+    """Manage the share structure for a business.
+
+    Assumption: The structure has already been validated, upon submission.
+
+    Other errors are recorded and will be managed out of band.
+    """
+    if not business or not share_structure:
+        # if nothing is passed in, we don't care and it's not an error
+        return None
+
+    err = []
+
+    # Delete the ones that are present in db but not in the json and create the ones in json but not in db.
+    if resolution_dates := share_structure.get('resolutionDates'):
+        # Two lists of dates in datetime format
+        business_dates = [item.resolution_date for item in business.resolutions]
+        parsed_dates = [parse(resolution_dt).date() for resolution_dt in resolution_dates]
+
+        # Dates in both db and json
+        inclusion_dates = [business.resolutions[index] for index, date in enumerate(business_dates)
+                           if date in parsed_dates]
+        if len(inclusion_dates) > 0:
+            business.resolutions = inclusion_dates
```

Comment:
i'm not sure what happens when you do this with respect to the versioning history.  In particular I would like to know if the unmodified `resolutions` entries result in a new `resolutions_version`.  If you can confirm the unmodified entries don't result in new version records,  you can do it this way.

Reply:
As per our conversation, we will test this out in DEV.  If versioning of records in the db is not correct, another PR will  be created to fix this.

Reply:
Sure. Thanks.